src/os_common/os_alloc.c

```diff
@@ -45,8 +45,9 @@ __wt_calloc(WT_SESSION_IMPL *session, size_t number, size_t size, void *retp)
 	 */
 	WT_ASSERT(session, number != 0 && size != 0);
 
-	if (session != NULL)
-		WT_STAT_FAST_CONN_INCR(session, memory_allocation);
+	/* Update statistics only if a connection has been established. */
```

Comment:
I think it's worth expanding on this comment to make it clear why it is important. Maybe "Only update statistics once we can be sure the connection and hence statistics tracking structures have been fully initialized."